test/python/dagcircuit/test_dagcircuit.py

```diff
@@ -739,6 +739,35 @@ def test_quantum_predecessors(self):
             or (isinstance(predecessor2, DAGInNode) and isinstance(predecessor1.op, Reset))
         )
 
+    def test_classical_predecessors(self):
+        """The method dag.classical_predecessors() returns predecessors connected by classical edges"""
+
+        self.dag.apply_operation_back(Measure(), [self.qubit1, self.clbit1], [])
+
+        predecessor_measure = self.dag.classical_predecessors(self.dag.named_nodes("measure").pop())
+
+        predecessor1 = next(predecessor_measure)
+
+        with self.assertRaises(StopIteration):
+            next(predecessor_measure)
+
+        self.assertTrue(isinstance(predecessor1, DAGInNode))
+        self.assertTrue(isinstance(predecessor1.wire, Clbit))
+
```

Comment:
With both of these tests, it'd be great if the circuit was slightly more complicated, and we asserted that we get the exact predecessors that we expect.  For example, you could use the same circuit you build in your release note example, then assert that the results come out in the way that you expect, including making sure everything's the correct node.

When you do `DAGCircuit.apply_operation_back`, the return value is the node that you need to pass into `classical_predecessors`, which might help write this test.

Reply:
ok, so I think is ok change the example with more gates, and keeping the outputs?

Reply:
I did  more complex quantum circuit and added the circuit diagram similar to other tests